File: source/OptimaJet.DWKit.StarterApplication/Models/SystemStatus.cs
using System;
using JsonApiDotNetCore.Models;

namespace OptimaJet.DWKit.StarterApplication.Models
{
    public class SystemStatus : Identifiable, ITrackDate, IBuildEngineReference
    {
        [Attr("build-engine-url")]
        public string BuildEngineUrl { get; set; }
        [Attr("build-engine-api-access-token")]
        public string BuildEngineApiAccessToken { get; set; }
        [Attr("system-available")]
        public bool SystemAvailable { get; set; }
        [Attr("date-created")]
        public DateTime? DateCreated { get; set; }
        [Attr("date-updated")]
        public DateTime? DateUpdated { get; set; }
    }
}

File: source/SIL.AppBuilder.Portal.Frontend/src/data/models/system-status.ts
import { AttributesObject, ResourceObject } from 'jsonapi-typescript';

export type SYSTEM_STATUSES_TYPE = 'systemStatus';
export const PLURAL_NAME = 'system-statuses';

export interface SystemStatusAttributes extends AttributesObject {
  buildEngineUrl: string;
  buildEngineApiAccessToken: string;
  systemAvailable?: boolean;
  dateCreated: string;
  dateUpdated: string;
}

export type SystemStatusResource = ResourceObject<SYSTEM_STATUSES_TYPE, SystemStatusAttributes>;
